framework/src/geomsearch/NearestNodeLocator.C

```diff
@@ -212,6 +233,100 @@ NearestNodeLocator::nearestNode(dof_id_type node_id)
   return _nearest_node_info[node_id]._nearest_node;
 }
 
+void
+NearestNodeLocator::updatePatch(std::vector<dof_id_type> & recheck_slave_nodes)
+{
+  Moose::perf_log.push("NearestNodeLocator::updatePatch()", "Execution");
+
+  std::vector<dof_id_type> trial_master_nodes;
+
+  // Build a bounding box.  No reason to consider nodes outside of our inflated BB
+  MeshTools::BoundingBox * my_inflated_box = NULL;
+
+  const std::vector<Real> & inflation = _mesh.getGhostedBoundaryInflation();
+
+  // This means there was a user specified inflation... so we can build a BB
+  if (inflation.size() > 0)
+  {
+    MeshTools::BoundingBox my_box = MeshTools::create_local_bounding_box(_mesh);
+
+    Real distance_x = 0;
+    Real distance_y = 0;
+    Real distance_z = 0;
+
+    distance_x = inflation[0];
+
+    if (inflation.size() > 1)
+      distance_y = inflation[1];
+
+    if (inflation.size() > 2)
+      distance_z = inflation[2];
+
+    my_inflated_box = new MeshTools::BoundingBox(Point(my_box.first(0) - distance_x,
+                                                       my_box.first(1) - distance_y,
+                                                       my_box.first(2) - distance_z),
+                                                 Point(my_box.second(0) + distance_x,
+                                                       my_box.second(1) + distance_y,
+                                                       my_box.second(2) + distance_z));
+  }
+
+  // Data structures to hold the Nodal Boundary conditions
+  ConstBndNodeRange & bnd_nodes = *_mesh.getBoundaryNodeRange();
+  for (const auto & bnode : bnd_nodes)
+  {
+    BoundaryID boundary_id = bnode->_bnd_id;
+    dof_id_type node_id = bnode->_node->id();
+
+    // If we have a BB only consider saving this node if it's in our inflated BB
+    if (!my_inflated_box || (my_inflated_box->contains_point(*bnode->_node)))
+    {
+      if (boundary_id == _boundary1)
+        trial_master_nodes.push_back(node_id);
+    }
+  }
+
+  // don't need the BB anymore
+  delete my_inflated_box;
+
+  // Convert trial master nodes to a vector of Points. This would be used to construct the KDTree.
+  std::vector<Point> master_points(trial_master_nodes.size());
+  for (unsigned int i = 0; i < trial_master_nodes.size(); ++i)
+  {
+    const Node & node = _mesh.nodeRef(trial_master_nodes[i]);
+    for (unsigned int j = 0; j < LIBMESH_DIM; ++j)
+      master_points[i](j) = node(j);
```

Comment:
Don't need the for loop. Just assign the points.

Reply:
Thanks for the comment. I have made this change.